beacon_node/network/src/network_beacon_processor/mod.rs

```diff
@@ -767,6 +769,73 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
                 "error" => %e)
         });
     }
+
+    /// Attempt to reconstruct all data columns if the following conditions satisfies:
+    /// - Our custody requirement is all columns
+    /// - We have >= 50% of columns, but not all columns
+    ///
+    /// Returns `Some(AvailabilityProcessingStatus)` if reconstruction is successfully performed,
+    /// otherwise returns `None`.
+    async fn attempt_data_column_reconstruction(
+        &self,
+        block_root: Hash256,
+    ) -> Option<AvailabilityProcessingStatus> {
+        let result = self.chain.reconstruct_data_columns(block_root).await;
+        match result {
+            Ok(Some((availability_processing_status, data_columns_to_publish))) => {
+                match &availability_processing_status {
+                    AvailabilityProcessingStatus::Imported(hash) => {
+                        debug!(
+                            self.log,
+                            "Block components available via reconstruction";
+                            "result" => "imported block and custody columns",
+                            "block_hash" => %hash,
+                        );
+                        self.chain.recompute_head_at_current_slot().await;
```

Comment:
Should we compute the head before publishing the columns? I don't have metrics but scheduling the columns for publishing should not block this thread for too long, and it's timing sensitive for the network to do so as soon as possible.

Reply:
Good catch! Recompute head should definitely happen after publish - publish is non blocking anyway.

Reply:
Fixed in https://github.com/sigp/lighthouse/pull/6403/commits/2e641908a83732dcf7279aaa35fd9fd3806e0d17